### PR TITLE
8282512: Remove usages of __WithField in compiler tests

### PR DESCRIPTION
**Pull request by dansmithcode**
Rewrote 2 compiler tests to perform 'withfield' operations in jasm code rather than using the '__WithField' javac hack.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8282512](https://bugs.openjdk.java.net/browse/JDK-8282512): Remove usages of __WithField in compiler tests

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/valhalla pull/664/head:pull/664` \
`$ git checkout pull/664`

Update a local copy of the PR: \
`$ git checkout pull/664` \
`$ git pull https://git.openjdk.java.net/valhalla pull/664/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 664`

View PR using the GUI difftool: \
`$ git pr show -t 664`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/valhalla/pull/664.diff">https://git.openjdk.java.net/valhalla/pull/664.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestDeoptimizationWhenBuffering.java [hunk 1]
```diff
@@ -74,35 +76,6 @@
...
-}
-
-final primitive class MyValue2 {
-    static int cnt = 0;
-    final int x;
-    public MyValue2() {
-        this.x = ++cnt;
-    }
-}
-
```
Comment (dansmithcode) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestDeoptimizationWhenBuffering.java:105]: Pretty directly lifted MyValue1 and MyValue2 into jasm code.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestWithfieldC1.java [hunk 2]
```diff
@@ -205,15 +98,7 @@ public void test1_verifier() {
...
-        FooValue v = FooValue.default;
-
-        v = __WithField(v.x, 1);
-        v = __WithField(v.y, 1);
-        foo_instance = v;
-
-        v = __WithField(v.x, 2);
-        v = __WithField(v.y, 2);
-        return v;
+        return FooValue.test2(this);
```
Comment (dansmithcode) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestWithfieldC1.java:101]: Moved this test into FooValue, with a parameter pointing back to 'this'.

File: test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestWithfieldC1.java [hunk 3]
```diff
@@ -310,21 +195,7 @@ public void test8_verifier() {
...
-        v = __WithField(v.x, 2);
-        v = __WithField(v.y, 2);
-
-        v2 = __WithField(v2.x, 3);
-        v2 = __WithField(v2.y, 3);
-
-        foo_instance = v2;
-        return v;
+        return FooValue.test9(this);
     }
```
Comment (dansmithcode) [inline test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestWithfieldC1.java:199]: Same—moved the code into FooValue, passed along a reference to 'this'.

**Review by TobiHartmann — APPROVED** [review]
That looks good to me, thanks for fixing!

Please make sure to run testing through tiers before integration.

**Pull request closed without merging**